azerothcore/azerothcore-wotlk
# fix(CORE/CREATURE) Polymorphed erratic mob behaviour

**Description:**
On the first polymorph cast on a creature, they are still trying to target the player.
This lasts the entire first cast of polymorph on a creature.
The sheep will stop moving and face the player after every spline path completes looking very strange.

There is many ways to fix this, but I went with the cleanest solution I feel like.

## Changes Proposed:
-  For creatures, if attack is called, and they are in polymorph form, return false.

## Issues Addressed:
https://github.com/azerothcore/azerothcore-wotlk/issues/5479

## Tests Performed:
- This builds without errors and works 100% 


## How to Test the Changes:
1. Start a gnome mage
2. grant level 60
3. grant all spells
4. Polymorph a boar or wolf
5. observe the strange stutter behavior is gone.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
